README.adoc

```diff
@@ -1,91 +1,511 @@
+sources contributed from our mounted volume.
+
+[source,json]
+----
+{
+   "name":"applicationConfig: [file:./config/application.properties]",
+   "properties":{
+      "server.shutdown":{
+         "value":"graceful",
+         "origin":"URL [file:./config/application.properties]:1:17"
+      },
+      "management.endpoints.web.exposure.include":{
+         "value":"*",
+         "origin":"URL [file:./config/application.properties]:2:43"
+      }
+   }
+}
+----
+
+Before continuing, be sure to stop the `port-forward` command.
+
+=== Service Discovery and Load Balancing
+
+For this part of the guide, you should install https://kustomize.io/[+++Kustomize+++]. Kustomize is a useful tool when
+working with Kubernetes and targeting different environments (dev, test, staging, production). We use it to generate
+YAML to deploy another application to Kubernetes that we will then be able to call using service discovery.
+
+Run the following command to deploy an instance of the https://github.com/ryanjbaxter/k8s-spring-workshop/tree/master/name-service[+++name-service+++]:
+
+```
+$ kustomize build "github.com/ryanjbaxter/k8s-spring-workshop/name-service/kustomize/multi-replica/" | kubectl apply -f -
+```
+
+This should deploy the `name-service` to your Kubernetes cluster. The deployment should create two replicas for the
+`name-service`:
+
+```
+$ kubectl get pods --selector app=k8s-workshop-name-service
+
+NAME                                         READY   STATUS    RESTARTS   AGE
+k8s-workshop-name-service-56b986b664-6qt59   1/1     Running   0          7m26s
+k8s-workshop-name-service-56b986b664-wjcr9   1/1     Running   0          7m26s
+
+```
+
+To demonstrate what this service does, we can make a request to it:
+
+```
+$ kubectl port-forward svc/k8s-workshop-name-service 9090:80
+
+$ curl http://localhost:9090 -i; echo
+
+HTTP/1.1 200
+k8s-host: k8s-workshop-name-service-56b986b664-6qt59
+Content-Type: text/plain;charset=UTF-8
+Content-Length: 4
+Date: Mon, 14 Sep 2020 15:37:51 GMT
+
+Paul
+```
+
+If you make multiple requests, you should see different names returned. Also note the header: `k8s-host`. This should
+align with the ID of the pod servicing the request.
+
+NOTE: When using the port-forwarding command, it makes only a request to a single pod, so you will only see one host
+in the response.
+
+Be sure to stop the `port-forward` command before moving on.
+
+With our service running, we can modify our application to make a request to the `name-service`.
+
+Kubernetes sets up DNS entries so that we can use the service ID for the `name-service` to make an HTTP request to the
+service without knowing the IP address of the pods. The Kubernetes service also load balances these requests between
+all the pods.
+
+In your application, open `DemoApplication.java` in `src/main/java/com/example/demo`. Modify the code as follows:
+
+[source,java]
+----
+include::app/src/main/java/com/example/demo/DemoApplication.java[]
```

Comment:
not rendering source code

Reply:
Yeah not on GitHub but I think it will once everything gets on Sagan.  It works fine in Intellij